lib/inspec/input.rb

```diff
@@ -98,15 +98,15 @@ def self.probe_stack
     # not been assigned a value. This allows a user to explicitly assign nil
     # to an input.
     class NO_VALUE_SET # rubocop: disable Naming/ClassAndModuleCamelCase
-      def initialize(name)
+      def initialize(name, obnoxiously_warn = true)
```

Comment:
This param might be better named `strict_warn` or `warn_if_not_set` 
And/or describe it in the comment lines before the method